soci/ztoc_test.go

```diff
@@ -236,53 +233,8 @@ func TestZtocGenerationConsistency(t *testing.T) {
 
 			// Compare raw Checkpoints
 			if !bytes.Equal(ztoc1.CompressionInfo.Checkpoints, ztoc2.CompressionInfo.Checkpoints) {
-
-				// compare Checkpoints within Go
-				index1, err := unmarshalGzipZinfo(ztoc1.CompressionInfo.Checkpoints[0])
-				if err != nil {
-					t.Fatalf("index from ztoc1 should contain data")
-				}
-				index2, err := unmarshalGzipZinfo(ztoc2.CompressionInfo.Checkpoints[0])
-				if err != nil {
-					t.Fatalf("index from ztoc2 should contain data")
-				}
-
-				if index1.have != index2.have {
-					t.Fatalf("index1.have=%d must be equal to index2.have=%d", index1.have, index2.have)
-				}
-
-				if index1.size != index2.size {
-					t.Fatalf("index1.size=%d must be equal to index2.size=%d", index1.size, index2.size)
-				}
-
-				if index1.spanSize != index2.spanSize {
-					t.Fatalf("index1.span_size=%d must be equal to index2.span_size=%d", index1.spanSize, index2.spanSize)
-				}
-
-				if len(index1.list) != len(index2.list) {
-					t.Fatalf("len(index1.list)=%d must be equal to len(index2.list)=%d", len(index1.list), len(index2.list))
-				}
-
-				for i := 0; i < len(index1.list); i++ {
-					indexPoint1 := index1.list[i]
-					indexPoint2 := index2.list[i]
-
-					if indexPoint1.bits != indexPoint2.bits {
-						t.Fatalf("index1.list[%d].bits=%d must be equal to index2.list[%d].bits=%d", i, index1.list[i].bits, i, index2.list[i].bits)
-					}
-
-					if indexPoint1.in != indexPoint2.in {
-						t.Fatalf("index1.list[%d].in=%d must be equal to index2.list[%d].in=%d", i, index1.list[i].in, i, index2.list[i].in)
-					}
-
-					if indexPoint1.out != indexPoint2.out {
-						t.Fatalf("index1.list[%d].out=%d must be equal to index2.list[%d].out=%d", i, index1.list[i].out, i, index2.list[i].out)
-					}
-
-					if !reflect.DeepEqual(indexPoint1.window, indexPoint2.window) {
-						t.Fatalf("index1.list[%d].window must be identical to index2.list[%d].window", i, i)
-					}
-				}
+				diffIdx := getPositionOfFirstDiffInByteSlice(ztoc1.CompressionInfo.Checkpoints, ztoc2.CompressionInfo.Checkpoints)
```

Comment:
Can just use `cmp.Diff` here.

Reply:
`cmp.Diff` can produce a lot of noise for checkpoints, since it produces the string difference of two slices in our case (https://github.com/google/go-cmp/blob/v0.5.9/cmp/compare.go#L113). The best would be to just return the index of the first difference and let investigate from here.

Reply:
Since this is only run in tests, I don't see any problem with the diff output, but that's fine I guess.